lib/gui/os/dialog/services/dialog.js

```diff
@@ -113,20 +113,28 @@ module.exports = function($q, SupportedFormatsModel) {
    * });
    */
   this.showWarning = (options) => {
+    const BUTTONS = [
+      options.confirmationLabel,
+      options.rejectionLabel
+    ];
+
+    const BUTTON_CONFIRMATION_INDEX = _.indexOf(BUTTONS, options.confirmationLabel);
+    const BUTTON_REJECTION_INDEX = _.indexOf(BUTTONS, options.rejectionLabel);
+
     return $q((resolve) => {
       electron.remote.dialog.showMessageBox(currentWindow, {
         type: 'warning',
         buttons: [
           options.confirmationLabel,
           options.rejectionLabel
         ],
-        defaultId: 1,
-        cancelId: 1,
+        defaultId: BUTTON_REJECTION_INDEX,
+        cancelId: BUTTON_REJECTION_INDEX,
         title: 'Attention',
         message: options.title,
         detail: options.description
       }, (response) => {
-        return resolve(response === 0);
+        return resolve(response === BUTTON_CONFIRMATION_INDEX);
```

Comment:
It means changing an integer-comparison to a string-comparison, but I guess `return resolve(BUTTONS[response] === options.confirmationLabel);` would work here?